src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -33,6 +33,7 @@ def get_requirement_strings(self):
     """
     opts = self.get_options()
     return (
+      "more-itertools<6.0.0 ; python_version<'3'",
```

Comment:
I think we'd also want to add the above TODO to this line to ensure they get changed in lockstep.